src/nvim/CMakeLists.txt

```diff
@@ -252,6 +268,17 @@ add_custom_command(OUTPUT ${GENERATED_OPTIONS}
   DEPENDS ${OPTIONS_GENERATOR} ${OPTIONS_LIST_FILE}
 )
 
+# Make NEOVIM_GENERATED_FOR_SOURCES and NEOVIM_GENERATED_FOR_HEADERS mutually exclusive
+# If a header in NEOVIM_GENERATED_FOR_HEADERS is also included in NEOVIM_GENERATED_SOURCES,
+# the header should be included only in NEOVIM_GENERATED_HEADERS
+
+foreach(hfile ${NEOVIM_GENERATED_FOR_HEADERS})
+  list(FIND NEOVIM_GENERATED_FOR_SOURCES ${hfile} hfile_idx)
+  if(NOT ${hfile_idx} EQUAL -1)
+    list(REMOVE_AT NEOVIM_GENERATED_FOR_SOURCES ${hfile_idx})
```

Comment:
Let me confirm. Shouldn't any generated file with the same name be included in both of headers and sources even in the future?